cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/ConfigurationGroupsCollection/index.tsx

```diff
@@ -94,30 +94,32 @@ const ConfigurationGroupsCollectionView: React.FC<ICreateContext & WithStyles<ty
       const newGroupID = 'ConfigGroup_' + uuidV4();
 
       // Add a new group's ID at the specified index
-      const newGroups = [...localConfigurationGroups];
-      if (newGroups.length === 0) {
-        newGroups.splice(0, 0, newGroupID);
+      let newGroups;
+      if (localConfigurationGroups.isEmpty()) {
+        newGroups = localConfigurationGroups.insert(0, newGroupID);
       } else {
-        newGroups.splice(index + 1, 0, newGroupID);
+        newGroups = localConfigurationGroups.insert(index + 1, newGroupID);
       }
       setLocalConfigurationGroups(newGroups);
 
       // Set the activeGroupIndex to the new group's index
-      if (newGroups.length <= 1) {
+      if (newGroups.size <= 1) {
         setActiveGroupIndex(0);
       } else {
         setActiveGroupIndex(index + 1);
       }
 
       // Set the mappings for the newly added group
-      setLocalGroupToInfo({
-        ...localGroupToInfo,
-        [newGroupID]: {
-          label: '',
-          description: '',
-        } as IConfigurationGroupInfo,
-      });
-      setLocalGroupToWidgets({ ...localGroupToWidgets, [newGroupID]: [] });
+      setLocalGroupToInfo(
+        localGroupToInfo.set(
+          newGroupID,
+          fromJS({
```

Comment:
it's better to be explicit when you know the type you want to convert (ie. use Map)

Reply:
Resolved.